subsys/bluetooth/mesh/shell/shell_sensor_cli.c

```diff
@@ -0,0 +1,450 @@
+
+	cadence_print(shell, err, &rsp);
+	return err;
+}
+
+static int cadence_set(const struct shell *shell, size_t argc, char *argv[], bool acked)
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SENSOR_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_sensor_cli *cli = mod->user_data;
+
+	uint32_t sensor_id = (uint16_t)strtol(argv[1], NULL, 0);
+	const struct bt_mesh_sensor_type *sensor_type = bt_mesh_sensor_type_get(sensor_id);
+	struct bt_mesh_sensor_cadence_status cadence = {
+		.fast_period_div = (uint8_t)strtol(argv[2], NULL, 0),
+		.min_int = (uint8_t)strtol(argv[3], NULL, 0),
+		.threshold = {
+			.delta = { .type = (int)strtol(argv[4], NULL, 0) },
+			.range = { .cadence = (int)strtol(argv[7], NULL, 0) }
+		}
+	};
+	struct bt_mesh_sensor_cadence_status rsp;
+
+	int err = shell_model_str2sensorval(argv[5], &cadence.threshold.delta.up);
+
+	if (err) {
+		return err;
+	}
+	err = shell_model_str2sensorval(argv[6], &cadence.threshold.delta.down);
+	if (err) {
+		return err;
+	}
+	err = shell_model_str2sensorval(argv[8], &cadence.threshold.range.low);
+	if (err) {
+		return err;
+	}
+	err = shell_model_str2sensorval(argv[9], &cadence.threshold.range.high);
+	if (err) {
+		return err;
+	}
+
+	if (acked) {
+		err = bt_mesh_sensor_cli_cadence_set(cli, NULL, sensor_type, &cadence, &rsp);
+		cadence_print(shell, err, &rsp);
+		return err;
+	} else {
+		return bt_mesh_sensor_cli_cadence_set_unack(cli, NULL, sensor_type, &cadence);
+	}
+}
+
+static int cmd_cadence_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	return cadence_set(shell, argc, argv, true);
+}
+
+static int cmd_cadence_set_unack(const struct shell *shell, size_t argc, char *argv[])
+{
+	return cadence_set(shell, argc, argv, false);
+}
+
+static void settings_print(const struct shell *shell, int err, uint16_t *ids, uint32_t count)
+{
+	if (!err) {
+		shell_fprintf(shell, SHELL_NORMAL, "[");
+		for (uint32_t i = 0; i < count; ++i) {
+			shell_fprintf(shell, SHELL_NORMAL, "0x%04x, ", ids[i]);
+		}
+		shell_print(shell, "]");
+	}
+}
+
+static int cmd_settings_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SENSOR_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_sensor_cli *cli = mod->user_data;
+
+	uint32_t count = CONFIG_BT_MESH_SHELL_SENSOR_CLI_MAX_SETTINGS;
+	uint16_t ids[CONFIG_BT_MESH_SHELL_SENSOR_CLI_MAX_SETTINGS];
+	uint32_t sensor_id = (uint16_t)strtol(argv[1], NULL, 0);
+	const struct bt_mesh_sensor_type *sensor_type = bt_mesh_sensor_type_get(sensor_id);
+
+	int err = bt_mesh_sensor_cli_settings_get(cli, NULL, sensor_type, ids, &count);
+
+	settings_print(shell, err, ids, count);
+	return err;
+}
+
+static void values_print(const struct shell *shell, int err, struct sensor_value *values,
+			 const struct bt_mesh_sensor_type *sensor_type)
+{
+	if (!err) {
+		shell_fprintf(shell, SHELL_NORMAL, "{ ");
+		for (uint8_t channel = 0; channel < sensor_type->channel_count; ++channel) {
```

Comment:
```suggestion
		for (uint32_t channel = 0; channel < sensor_type->channel_count; ++channel) {
```

Reply:
What? `bt_mesh_sensor_type::channel_count` is `uint8_t`

Reply:
Register size is 4 bytes, no point to use `uint8_t` here.